PYIC-7964: Reinstate core-front backend response logging

AmritSidhu opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Reinstate core-front backend response logging
Exclude data logging for the proven-identity-details request only

### Why did it change

In core-front we were logging the response data we get back for all core-back requests. This is useful for debugging as it includes for example CRI ids, redirect URIs and page/journey responses. Unfortunately we discovered we were inadvertently logging PII (name/dob/address) for the proven-identity-details request, and the quick incident fix was just to drop the data property for all API requested completed logs

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-7964](https://govukverify.atlassian.net/browse/PYIC-7964)

## Checklists

### Environment variables or secrets

<!-- Delete if changes DO include new environment variables or secrets -->
- [x] No environment variables or secrets were added or changed


[PYIC-7964]: https://govukverify.atlassian.net/browse/PYIC-7964?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AmritSidhu committed:
Revert "BAU Stop logging proven-identity-details PII"

This reverts commit 8e204c85dbb46b2bdabf3c400846137401ba662d.

AmritSidhu committed:
PYIC-7964: Reinstate core-front backend response logging

establish an ‘allow list’ of requests to only include non-PII data logging

blakeyp commented on src/app/shared/axiosHelper.ts:
```diff
@@ -34,37 +33,70 @@ const extractCredentialIssuerId = (
+      const body = { ...response.data };
+
+      const endpoint = response.config?.url ?? "";
+      if (endpoint.includes("proven-identity-details")) {
```
Could we make this an allow list rather than deny list ie. specify which requests we should allow data logging for (which I think should be all except proven-identity-details)? so we don't inadvertently start leaking PII if we add new requests in the future

Also could we pull the values from config eg. `API_SESSION_INITIALISE` rather than hardcoding?

AmritSidhu committed:
PYIC-7964: Use an allow list rather than deny list

specify which requests we should allow data logging for this so we don't inadvertently start leaking PII if we add new requests in the future

blakeyp commented on src/app/shared/axiosHelper.ts:
```diff
@@ -12,13 +13,25 @@ declare module "axios" {
+// Helper function to determine if an endpoint is in the allow list
+const isEndpointAllowedForDataLogging = (url: string): boolean => {
+  const ALLOWED_ENDPOINTS = Object.values(API_URLS).filter(
+    (endpoint) => endpoint !== API_URLS.API_BUILD_PROVEN_USER_IDENTITY_DETAILS,
```
Hmm to me this is still a deny list as we're allowing anything except this request - I think we want to explicitly define the allowed requests? so that if we introduce a new API endpoint in the future, we need to explicitly add it to the allowlist here - otherwise we may start calling a new endpoint that logs PII without us realising

blakeyp commented on src/constants/api-constants.ts:
```diff
@@ -0,0 +1,9 @@
+  API_SESSION_INITIALISE: "/session/initialise",
+  API_BUILD_PROVEN_USER_IDENTITY_DETAILS: "/user/proven-identity-details",
+});
+
```
could we pull these directly from config.ts rather than having to redefine them here?

AmritSidhu committed:
PYIC-7964: Use constants from config.ts, create a specific allowList in the logger

blakeyp commented on src/app/shared/axiosHelper.ts:
```diff
@@ -22,9 +22,18 @@ interface RequestLog {
   duration?: number;
 }
 
+const ALLOWED_ENDPOINTS: string[] = [
```
could name this is a bit more explicitly eg. `ALLOWED_RESPONSE_DATA_LOGGING_ENDPOINTS` ?

AmritSidhu committed:
PYIC-7964: Update allowed list naming
